hyperspy/io_plugins/impulse.py

```diff
@@ -0,0 +1,207 @@
+import numpy as np
+import os
+from datetime import datetime as dt
+import csv
+import logging
+
+
+# Plugin characteristics
+# ----------------------
+format_name = "Impulse"
+description = "Reads DENSsolutions Impulse log files"
+full_support = False
+# Recognised file extension
+file_extensions = ["csv", "CSV", "impulse"]
+default_extension = 0
+# Reading capabilities
+reads_images = False
+reads_spectrum = False
+reads_spectrum_image = False
+# Writing capabilities
+writes = False
+non_uniform_axis = False
+# ----------------------
+
+
+_logger = logging.getLogger(__name__)
+
+
+# At some point, if there is another readerw, whith also use csv file, it will
+# be necessary to mention the other reader in this message (and to add an
+# argument in the load function to specify the correct reader)
+invalid_file_error = (
+    "The csv reader can't import the file, please"
+    " make sure, that this is a valid Impulse log file."
+)
+invalid_filenaming_error = {
+    "The filename does not match Impulse naming, please"
+    " make sure that the filenames for the logfile and metadata file are unchanged."
+}
+no_metadata_file_error = {
+    "No _Metadata file detected in filepath, please"
+    " make sure that the Metadata file is included."
+}
+
+
+def file_reader(filename, *args, **kwds):
+    csv_file = impulseCSV(filename)
+    return _impulseCSV_log_reader(csv_file)
+
+
+def _impulseCSV_log_reader(csv_file):
+    csvs = []
+    for key in csv_file.logged_quantity_name_list:
+        try:
+            csvs.append(csv_file.get_dictionary(key))
+        except BaseException:
+            raise IOError(invalid_file_error)
+    return csvs
+
+
+class impulseCSV:
```

Comment:
Python class name follows CapWords convention.
```suggestion
class ImpulseCSV:
```